src/test/java/com/auction/repositoryTest/BidRepositoryTest.java

```diff
@@ -0,0 +1,211 @@
+    @Autowired
+    private AuctionRepository auctionRepository;
+    @Test
+    @Rollback(false)
+    public void testSaveBid() {
+        User user = new User();
+        user.setName("Bidder One");
+        user.setPassword("password123");
+        user.setPhoneNumber("123456789");
+        user.setAddress("123 Main St");
+        user = userRepository.save(user);
+        Product product = new Product();
+        product.setName("Bicycle");
+        product.setStartingPrice(200.0f);
+        product.setSellPrice(0.0f);
+        product.setStatus("AVAILABLE");
+        product.setCategory("Sports");
+        product.setSeller(user);
+        product = productRepository.save(product);
+        Auction auction = new Auction();
+        auction.setStartDate(new Date());
+        auction.setEndDate(new Date(System.currentTimeMillis() + 86400000));
+        auction.setProduct(product);
+        auction.setStatus("ONGOING");
+        auction = auctionRepository.save(auction);
+        Bid bid = new Bid();
+        bid.setValue(250.0f);
+        bid.setTimeStamp(new Date());
+        bid.setUser(user);
+        bid.setProduct(product);
+        bid.setAuction(auction);
+        Bid savedBid = bidRepository.save(bid);
+        assertNotNull(savedBid);
+        assertTrue(savedBid.getId() > 0);
+        assertEquals(250.0f, savedBid.getValue());
+        assertEquals(user.getId(), savedBid.getUser().getId());
+        assertEquals(product.getId(), savedBid.getProduct().getId());
+        assertEquals(auction.getId(), savedBid.getAuction().getId());
+    }
+    @Test
+    public void testFindBidById() {
+        User user = new User();
+        user.setName("Bidder Two");
+        user.setPassword("password123");
+        user.setPhoneNumber("987654321");
+        user.setAddress("456 Another St");
+        user = userRepository.save(user);
+        Product product = new Product();
+        product.setName("Laptop");
+        product.setStartingPrice(500.0f);
+        product.setSellPrice(0.0f);
+        product.setStatus("AVAILABLE");
+        product.setCategory("Electronics");
+        product.setSeller(user);
+        product = productRepository.save(product);
+        Auction auction = new Auction();
+        auction.setStartDate(new Date());
+        auction.setEndDate(new Date(System.currentTimeMillis() + 86400000));
+        auction.setProduct(product);
+        auction.setStatus("ONGOING");
+        auction = auctionRepository.save(auction);
+        Bid bid = new Bid();
+        bid.setValue(600.0f);
+        bid.setTimeStamp(new Date());
+        bid.setUser(user);
+        bid.setProduct(product);
+        bid.setAuction(auction);
+        bid = bidRepository.save(bid);
+        Optional<Bid> foundBid = bidRepository.findById(bid.getId());
+        assertTrue(foundBid.isPresent());
+        assertEquals(600.0f, foundBid.get().getValue());
+    }
+    @Test
+    public void testFindAllBids() {
+        User user = new User();
```

Comment:
In such cases it is better to have some common method that will generate required object (User) and reuse it